app/controllers/hosts_controller.rb

```diff
@@ -629,14 +631,35 @@ def preview_host_collection
     end
   end
 
+  def get_statuses
```

Comment:
This is bad action name. We should not use `get_` prefixes.
Would `statuses` be enough?